### PR TITLE
[Development] Add dataset generation and config specifically for capabilities.

### PR DESCRIPTION
**Pull request by Zhouxunzhe**
## Motivation and Context
Add dataset generation and config specifically for capabilities.

## How Has This Been Tested
Tested locally.

## Types of changes
\[Development\]

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Dependency Upgrade\]** Upgrades one or several dependencies in habitat
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!
- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 1]
```diff
@@ -1255,7 +1269,122 @@ def set_receptacle_in_scene(self, ep_info):
                 np.min(bounds, axis=0), np.max(bounds, axis=0)
             )
         return receps
-   
+
+    def set_desired_receptacle_in_scene(self, ep_info, receptacle_name):
```
Comment (Tradewindycc) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py]: Maybe we could merge this function with `set_receptacles_in_scene`, they function basically the same.

**Review by Tradewindycc — COMMENTED** [review]
LGTM!

File: habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py [hunk 2]
```diff
@@ -984,7 +989,8 @@ def get_navigable_receptacles(
                     sim=sim,
                     point=point,
                     height=max_access_height,
-                    nav_to_min_distance=nav_to_min_distance,
+                    # nav_to_min_distance=nav_to_min_distance,
+                    nav_to_min_distance=-1,
```
Comment (Tradewindycc) [inline habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py]: By the way, we'd better revert this change.

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 3]
```diff
@@ -760,11 +760,27 @@ def _add_objs(
...
+            recep_to_remove = rom.get_object_by_handle(old_recep_handle)
+            # save the recep to move
+            move_recep_template = otm.get_templates_by_handle_substring(recep_to_remove.handle[:-6])
+            move_recep_template = list(move_recep_template.keys())[0]
+            self.prev_remove_recep = move_recep_template
+
+            rom.remove_object_by_id(recep_to_remove.object_id)
+            receps = self.set_receptacle_in_scene(ep_info, "wall_cabinet")
+            self._receptacles_cache[ep_info.scene_id] = receps
+            self._receptacles = receps
```
Comment (SgtVincent) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py]: Is this code used to add shelves into hssd scene?

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py [hunk 4]
```diff
@@ -1162,81 +1178,121 @@ def get_runtime_perf_stats(self) -> Dict[str, float]:
...
+                        f"Template not found for object with handle {tar_recep_handle}"
+                    )
+                tar_path = list(tar_template.keys())[0]
+                with open(tar_path) as tar_file:
+                    tar_object_config = json.load(tar_file)
+                user_defined = tar_object_config['user_defined']
+                if 'scale' in user_defined:
+                    tar_scale = user_defined['scale']
+                else:
+                    tar_scale = user_defined[list(user_defined.keys())[0]]['scale']
```
Comment (SgtVincent) [inline habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:1249]: Do we also randomize the target object scale when inserting new objects into the scene? Why do we need this?

Comment (Zhouxunzhe) [reply habitat-lab/habitat/tasks/rearrange/rearrange_sim.py:1249]: Since the scales in replica dataset and hssd dataset are different, to set the wall cabinet (which is a receptacle in replica dataset) properly, we need adjust the scale of it. 
For the question about the random scale, the simulator doesn't randomly init the scale either. It just set the scale as (1,1,1).

File: habitat-lab/habitat/tasks/rearrange/rearrange_sim.py (hunk 3 above)
Comment (Tradewindycc) [reply habitat-lab/habitat/tasks/rearrange/rearrange_sim.py]: Yep, we were just debugging this code snippet together. @Zhouxunzhe

File: habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py (hunk 2 above)
Comment (Zhouxunzhe) [reply habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py]: solved :) Thank you for reminding me!

**Comment by Zhouxunzhe** [comment]
In the latest commit (607544dc92bb46b4f117067350728c3392ca1ddd), I revert the initial version of rearrange generator. And I add hssd generator as individual files.

**Review by Tradewindycc — APPROVED** [review]
LGTM!
